src/types.ts

```diff
@@ -210,6 +226,11 @@ export type PstInput = {
   status: string;
 };
 
+export type DelayedEvolveInput = {
+  contractSrcTxId: string; // The source code that this contract will evolve to
+  evolveHeight?: number; // The height at which this evolution action takes effect
```

Comment:
is this optional as it could be immediately/the current block height?

Reply:
It is optional and if the submittor doesnt provide it then its set to +SmartWeave.block.height + MINIMUM_ALLOWED_EVOLUTION_DELAY;